### ISSUE TITLE
Using range operator causes InvalidProgramException

### ISSUE THREAD

**Issue by vinaychandra** [dotnet/roslyn#35142]
# Issue Title

Using range operator causes InvalidProgramException in the latest dotnet core preview

# General

Donet Core Version: 3 Preview 4
Visual Studio: 16.0.2

Running an MsTest v2 that targets netcore3.0 which tests another netcore3.0 library that uses the new range operator causes an `InvalidProgramException`

Example code used in the library:
`var index = query[0..^1] + (char)(query[^1] + 1);`

Example that worked
`var index = query.Substring(0, query.Length - 1) + query[query.Length - 1];`

Exception:
```
Exception Message: Common Language Runtime detected an invalid program.
```

**Comment by agocke** [comment]
@terrajobst Did the compiler I inserted into `buildtools` and `arcade` flow into .net core yet?

**Comment by agocke** [comment]
I can't reproduce this on Roslyn master: https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4AEAMACFBGA3AWACgUAmDAYSIG8iM6sBmLbANiwBYMBZCASwDsAFAEpa9GoXpTuggEQRgAY1nCCk+gF8xdFExxsUnLoJyYAjgFdEAT1Hq6E6fQBuEBBn5wAHjACS/ABNvDABeDEsbAG00ADoYgD1sAF0MAGoMQUUACzdhQQiEa0jElPTsVW1pHABOQU8ffyCvCvsMLUJ2on4IMDgoAAcIRThmakr0ZhiAJQt+GF5emPIAezB+3gAbRABlRGdeYag1KV0MBDgIAOX+DesMWAQLRRgMRuCQV4BRSwgYBS2ADxvLwAPkqjicGH6CF4rngZwuVxudwELwA+q4NlY1JUpJEuHAYFllgFfGsNoICUSSWT+hsAPL9ebXKAxACCAHMOecoFBYXB/BsBAIOcIkrj6KdgYJURhMVYADQYYDLZYbDAAMwQq0+gVCmogGygcDskIwELNdF4Goy8pGAIwaFNlocEpdRO1AHcPHBvWyEByLL05vSLDB6RqphB+By4J8vMMmbxriJji62kQ3dJrRktTrAs70/QMYarPqAH52tMuuBGuBZs0lrEjMJVrOdVpSAD0XYwABUsiMNWqNstPSKoTC4SNFCyYI9nssEFAMJABLcMIpzr84AFNUuoYgbduoNcMDBlhgIIZz4PN4PFABrKBZ6GwnevQLeGVzOWlk1ZhalpNmWrb/tW0gdmaPa9gCUBBpAhQguQ27wtGn5NFCyyohOvyarwy4vLWcDBjAMQAl28FgIh1hgp2kp6KwGHBNsfwIC8IQgj63rSk6OL0XQMEYHBCFuLRKEXGh/DMV4WE4TGV4vMAcDWNce4bNARFbKR5GUaJSFZqc+gyRgup7px3EyYI5Z8ZmAkYPihLEqS5KUk5NLkoyzL8KynLcn0fLOAKNzCjGYqGYxbDAhgABi2pgKxbgwD+Lx2oW0hAWaOaCHawmOulZqZemHpjpZ/qBqRobhpG0axvGibeamDaaHZRYoAA7JZ0ppRBUhQZCjnUi5dJuUNtIMkmLLslyPKBcFQr8CK4X2UZTHRXF+YBClf7NgV4jNVaNo5f+eVOgd5rnVIJXep4foBkGcAhmGEZRjGcYJnAk1CC0Rb9S6HVdV+XjWT17atZCQkidRYkglMhIWAgPm3iMAiYXaulUTRdFmqcsoAGr/oB52xjA51FS62Ugfa+WXZCAPllTvXprWxq004AOM+d7Rmn9VRMCqaqvFAG1gGZoRcVTp38Wag3OeNo1y55X2+TNAX8oKoWiuKK1MLKADihIRhqxrJbKWwxkSe2uvZUiyssGrG4S0vptlvjC/FZlW5a9uOxxGDmxyRIYAAtBkDOgzb9As/WkdOD7Jv6pzhDnQD8dO2Dyc6xgyxBQgMJBMqqrqt8FiGlAgjLMAABWcDPDtVjCOL9coyu0UAGRtxgkshGEgiCMCwhpTEScRYXgsl2XGTRcsRKII3Fnd2EM+Dggw9tpnONMDniD5yjv4GzAAAS0BZCsQQiE3I9Z8ZCx0gcvAvMs/SIL8B7SrKaVNyLiXscdu0Qe0Ds3RegDCGCMHAYxWgTBwNMWY8xFgrDWJsHYewDh9GdjoJg24kQbgeE8F4r1YzglHtFH+LwqAYBJrgDMG86ZMGimLChVCaGj0IXAfuQN7hsRgEqaKj0Ahe3JtIMh+pYBJSZk4MWYR+ESJYdfJibCMBkLZMlaKYj2Lzy4rdDAbDBDqJ4TJGInsMF82YGwRRZkVEcMwvwzRlldHAhiGQpUtiTEnEijo2qIw2QbHVBZbRDigZOO4bwoJxjwZOFlh5EaVJFZ0i8smHy01/K8nViFRaYVtabwwEbE2bJAgABlHqByyBgA+uTCT5ICEUi2WRtoB0tkTWOGBZT6NkVINR3DophDIe0+g2V9GOLdiLT2bMuFJX1A00podBlBIJs2PpdBo5jP0aIrpczwIRMtLKGR50+GBG6aZQIiyWlHX4UM92m0vZmn4ZM4pQdQ7nI2Qs86yzmlSFudIg5zzsRbKykdQQFhUTCFuVxQFwKpkYCQEgDIQK5jCFWWCuFMAQUFjJmM66ZV7qVWejVN69VPqNR+umXmONOraIqTAKpNSSl6JCRgW5MzuHEqcKShiCJLjXFwfOfBOSHZ5MKfcrITS2q61/JS80lCnY0NFS038NKg5MOlWy9mTBKXUqFdtNOBizZCuuftd59AJVL35enQ1dAFWlLCFMk5KrTE3gACK1znAuZKywwxysfqagx7qXi6tqfq62RYpDav1JSk5UhIVhEtbajOfUiAaCAA==

**Comment by terrajobst** [comment]
> @terrajobst Did the compiler I inserted into `buildtools` and `arcade` flow into .net core yet?

Not sure. @joperezr or @wtgodbe would know. However, given a customer reported this I'm pretty sure it's the Preview 3 or 4 compiler. Sounds like the new compiler no longer has the bug. Might still be worth root causing though. It could be that the new thing binds differently and doesn't exhibit this behavior for built-in types.

**Comment by wtgodbe** [comment]
@agocke can you link me the commits where you inserted the new compiler into buildtools/arcade? I can check if those changes have flowed into core.

**Comment by agocke** [comment]
PR was https://github.com/dotnet/arcade/pull/2536, I'm still trying to figure out what build has this bug

**Comment by wtgodbe** [comment]
That Arcade fix has been in CoreFx since https://github.com/dotnet/corefx/pull/36997

**Comment by canton7** [comment]
Related: #34209 and #34840, fixed by #33679?

**Comment by gafter** [comment]
We believe this has been fixed in the latest previews.  Please let us know if that is not the case.
